tests/unit/test_charm_scaling.py

```diff
@@ -165,6 +163,33 @@ def test_scaling_instance_map_configured(harness: Harness) -> None:
                 "port": 8034,
             },
         }
+    worker_config_path = root / synapse.SYNAPSE_WORKER_CONFIG_PATH[1:]
+    with open(worker_config_path, encoding="utf-8") as config_file:
+        content = yaml.safe_load(config_file)
+        assert content["worker_name"] == "federationsender1"
+
+
+def test_scaling_instance_restarts_federation_service(
+    harness: Harness, monkeypatch: pytest.MonkeyPatch
+) -> None:
+    """
+    arrange: charm deployed, integrated with Redis, one more unit in peer relation
+        and set as leader.
+    act: emit config-changed event.
+    assert: Synapse charm is configured with instance_map and the federation service is restarted.
+    """
+    rel_id = harness.add_relation(synapse.SYNAPSE_PEER_RELATION_NAME, "synapse")
+    harness.add_relation_unit(rel_id, "synapse/1")
+    harness.begin_with_initial_hooks()
+    federation_container = harness.model.unit.containers[synapse.SYNAPSE_CONTAINER_NAME]
+    harness.set_can_connect(federation_container, True)
+    harness.add_relation("redis", "redis", unit_data={"hostname": "redis-host", "port": "1010"})
+    harness.set_leader(True)
+
+    restart_federation_mock = MagicMock()
+    monkeypatch.setattr(pebble, "restart_federation_sender", restart_federation_mock)
+    harness.update_config({"workers_ignore_list": "worker1"})
```

Comment:
Is this configuration needed?

Reply:
I think it's used to trigger a restart because that's what we need for this test (see "act")